src/auth/OnlyContract.huff

```diff
@@ -0,0 +1,14 @@
+/// @title OnlyContract
+/// @author asnared <https://github.com/abigger87>
+
+/* Reverts if the caller is not a contract using tx.origin */
+/* WARNING: This is an anti-pattern and prevents interoperability */
+/* WARNING: Will need to change once account abstraction is implemented */
```

Comment:
Like you said, it's an anti-pattern but not solely because of account abstraction assuming we're talking of EIP-2938. [EIP-3074](https://eips.ethereum.org/EIPS/eip-3074) is another concern as it essentially allows users to delegate control of an EOA to a contract. Additionally, users of wallets like argent, authereum and gnosis multi-sigs are able to "bypass" this access control.

I think these risks should be documented alongside the above so devs fully understand what they're getting themselves into.

Reply:
Good catch, will do!